### PR TITLE
Don't transpile as much on desktop

### PR DESCRIPTION
**Pull request by chrisnojima**
This changes our webpack config so on electron we transpile the minimal set through babel.

this allows us to
- [x] Put breakpoints in generators (!!!) and see useful stack traces there
- [x] Hopefully better performance (didn't test this)

SADLY, we can't turn this on in production (yet). 
Uglify doesn't support modern JS so it chokes on the input
Babili is way too immature (lots of github issues about it generating incorrect code :( ) and it crashes when I try a production build

So we basically undo all this cool stuff on a prod build...

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits b6028edc0b42...4fe8ed7d26bb introduce visual changes on linux. <details><summary>:mag_right: 6 changed</summary>

 * Changed: **Markdown-special_chars_in_code_block** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490890971597-b6028edc0b42-4fe8ed7d26bb-linux/Markdown-special_chars_in_code_block-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490890971597-b6028edc0b42-4fe8ed7d26bb-linux/Markdown-special_chars_in_code_block-4fe8ed7d26bb.png)  
   Markdown-special_chars_in_code_block comparison
 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490890971597-b6028edc0b42-4fe8ed7d26bb-linux/Markdown-links-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490890971597-b6028edc0b42-4fe8ed7d26bb-linux/Markdown-links-4fe8ed7d26bb.png)  
   Markdown-links comparison
 * Changed: **Markdown-Quotes** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490890971597-b6028edc0b42-4fe8ed7d26bb-linux/Markdown-Quotes-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490890971597-b6028edc0b42-4fe8ed7d26bb-linux/Markdown-Quotes-4fe8ed7d26bb.png)  
   Markdown-Quotes comparison
 * Changed: **Markdown-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490890971597-b6028edc0b42-4fe8ed7d26bb-linux/Markdown-Normal-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490890971597-b6028edc0b42-4fe8ed7d26bb-linux/Markdown-Normal-4fe8ed7d26bb.png)  
   Markdown-Normal comparison
 * Changed: **Markdown-Code_block** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490890971597-b6028edc0b42-4fe8ed7d26bb-linux/Markdown-Code_block-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490890971597-b6028edc0b42-4fe8ed7d26bb-linux/Markdown-Code_block-4fe8ed7d26bb.png)  
   Markdown-Code_block comparison
 * Changed: **Markdown-Blank_lines** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490890971597-b6028edc0b42-4fe8ed7d26bb-linux/Markdown-Blank_lines-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490890971597-b6028edc0b42-4fe8ed7d26bb-linux/Markdown-Blank_lines-4fe8ed7d26bb.png)  
   Markdown-Blank_lines comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits b6028edc0b42...396b368b29c7 introduce visual changes on linux. <details><summary>:mag_right: 6 changed</summary>

 * Changed: **Markdown-special_chars_in_code_block** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490895043402-b6028edc0b42-396b368b29c7-linux/Markdown-special_chars_in_code_block-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490895043402-b6028edc0b42-396b368b29c7-linux/Markdown-special_chars_in_code_block-396b368b29c7.png)  
   Markdown-special_chars_in_code_block comparison
 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490895043402-b6028edc0b42-396b368b29c7-linux/Markdown-links-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490895043402-b6028edc0b42-396b368b29c7-linux/Markdown-links-396b368b29c7.png)  
   Markdown-links comparison
 * Changed: **Markdown-Quotes** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490895043402-b6028edc0b42-396b368b29c7-linux/Markdown-Quotes-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490895043402-b6028edc0b42-396b368b29c7-linux/Markdown-Quotes-396b368b29c7.png)  
   Markdown-Quotes comparison
 * Changed: **Markdown-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490895043402-b6028edc0b42-396b368b29c7-linux/Markdown-Normal-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490895043402-b6028edc0b42-396b368b29c7-linux/Markdown-Normal-396b368b29c7.png)  
   Markdown-Normal comparison
 * Changed: **Markdown-Code_block** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490895043402-b6028edc0b42-396b368b29c7-linux/Markdown-Code_block-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490895043402-b6028edc0b42-396b368b29c7-linux/Markdown-Code_block-396b368b29c7.png)  
   Markdown-Code_block comparison
 * Changed: **Markdown-Blank_lines** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490895043402-b6028edc0b42-396b368b29c7-linux/Markdown-Blank_lines-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490895043402-b6028edc0b42-396b368b29c7-linux/Markdown-Blank_lines-396b368b29c7.png)  
   Markdown-Blank_lines comparison
</summary>

**Comment by cjb** [comment]
Huh, wonder what's up with the visdiff.

**Comment by chrisnojima** [comment]
yeah its strange. i just put a workaround to a bug in babel so maybe that'll help. in app the visdiff is correct on these screens

File: shared/.babelrc [hunk 1]
```diff
@@ -15,26 +15,6 @@
     ]
   ],
   "env": {
-    // Used to build electron app
```
Comment (chrisnojima) [inline shared/.babelrc]: The electron webpack build ignores babelrc files. The way babel-loader works is it'll inherit parent babelrc files so what we want to do w/ RN and our scripts 'infects' the build output

File: shared/desktop/webpack.config.base.js [hunk 2]
```diff
@@ -21,7 +21,32 @@ module.exports = {
...
+        // Have to do this or it'll inherit babelrcs from the root and pull in things we don't want
+        babelrc: false,
+        presets: [
+          ['env', {
+            useBuiltIns: false,
+            targets: {
+              electron: '1.4.12',
+            },
+            debug: true,
+            exclude: ['transform-regenerator'],
```
Comment (chrisnojima) [inline shared/desktop/webpack.config.base.js]: we do not want regenerator being included

File: shared/desktop/webpack.config.base.js [hunk 3]
```diff
@@ -21,7 +21,32 @@ module.exports = {
...
+          }],
+          'babel-preset-react',
+        ],
+        plugins: [
+          [
+            'babel-plugin-transform-builtin-extend',
+            {globals: ['Error']},
+          ],
+          'transform-flow-strip-types',
+          'transform-object-rest-spread', // not supported by electron yet
```
Comment (chrisnojima) [inline shared/desktop/webpack.config.base.js]: add back some stage-2 stuff we use

**Comment by keybase-ci-visdiff** [comment]
The commits b6028edc0b42...875502e0a4ed introduce visual changes on linux. <details><summary>:mag_right: 6 changed</summary>

 * Changed: **Markdown-special_chars_in_code_block** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490899213517-b6028edc0b42-875502e0a4ed-linux/Markdown-special_chars_in_code_block-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490899213517-b6028edc0b42-875502e0a4ed-linux/Markdown-special_chars_in_code_block-875502e0a4ed.png)  
   Markdown-special_chars_in_code_block comparison
 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490899213517-b6028edc0b42-875502e0a4ed-linux/Markdown-links-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490899213517-b6028edc0b42-875502e0a4ed-linux/Markdown-links-875502e0a4ed.png)  
   Markdown-links comparison
 * Changed: **Markdown-Quotes** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490899213517-b6028edc0b42-875502e0a4ed-linux/Markdown-Quotes-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490899213517-b6028edc0b42-875502e0a4ed-linux/Markdown-Quotes-875502e0a4ed.png)  
   Markdown-Quotes comparison
 * Changed: **Markdown-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490899213517-b6028edc0b42-875502e0a4ed-linux/Markdown-Normal-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490899213517-b6028edc0b42-875502e0a4ed-linux/Markdown-Normal-875502e0a4ed.png)  
   Markdown-Normal comparison
 * Changed: **Markdown-Code_block** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490899213517-b6028edc0b42-875502e0a4ed-linux/Markdown-Code_block-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490899213517-b6028edc0b42-875502e0a4ed-linux/Markdown-Code_block-875502e0a4ed.png)  
   Markdown-Code_block comparison
 * Changed: **Markdown-Blank_lines** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490899213517-b6028edc0b42-875502e0a4ed-linux/Markdown-Blank_lines-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490899213517-b6028edc0b42-875502e0a4ed-linux/Markdown-Blank_lines-875502e0a4ed.png)  
   Markdown-Blank_lines comparison
</summary>

File: shared/desktop/yarn-helper.js [hunk 4]
```diff
@@ -43,19 +43,19 @@ const commands = {
     help: 'Copy font output into the right folders',
   },
   'build-dev': {
-    env: {BABEL_ENV: 'electron', NO_SERVER: 'true'},
+    env: {BABEL_ENV: 'yarn', NO_SERVER: 'true'},
```
Comment (chrisnojima) [inline shared/desktop/yarn-helper.js]: there is no electron BABEL_ENV anymore

**Comment by chrisnojima** [comment]
hmmmm. visdiff still messed. @chromakode ?

**Comment by chrisnojima** [comment]
- [ ] test prod build

**Comment by keybase-ci-visdiff** [comment]
The commits b6028edc0b42...74f52b743b88 introduce visual changes on linux. <details><summary>:mag_right: 6 changed</summary>

 * Changed: **Markdown-special_chars_in_code_block** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490900144612-b6028edc0b42-74f52b743b88-linux/Markdown-special_chars_in_code_block-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490900144612-b6028edc0b42-74f52b743b88-linux/Markdown-special_chars_in_code_block-74f52b743b88.png)  
   Markdown-special_chars_in_code_block comparison
 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490900144612-b6028edc0b42-74f52b743b88-linux/Markdown-links-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490900144612-b6028edc0b42-74f52b743b88-linux/Markdown-links-74f52b743b88.png)  
   Markdown-links comparison
 * Changed: **Markdown-Quotes** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490900144612-b6028edc0b42-74f52b743b88-linux/Markdown-Quotes-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490900144612-b6028edc0b42-74f52b743b88-linux/Markdown-Quotes-74f52b743b88.png)  
   Markdown-Quotes comparison
 * Changed: **Markdown-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490900144612-b6028edc0b42-74f52b743b88-linux/Markdown-Normal-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490900144612-b6028edc0b42-74f52b743b88-linux/Markdown-Normal-74f52b743b88.png)  
   Markdown-Normal comparison
 * Changed: **Markdown-Code_block** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490900144612-b6028edc0b42-74f52b743b88-linux/Markdown-Code_block-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490900144612-b6028edc0b42-74f52b743b88-linux/Markdown-Code_block-74f52b743b88.png)  
   Markdown-Code_block comparison
 * Changed: **Markdown-Blank_lines** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490900144612-b6028edc0b42-74f52b743b88-linux/Markdown-Blank_lines-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490900144612-b6028edc0b42-74f52b743b88-linux/Markdown-Blank_lines-74f52b743b88.png)  
   Markdown-Blank_lines comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits b6028edc0b42...34da97390f31 introduce visual changes on linux. <details><summary>:mag_right: 6 changed</summary>

 * Changed: **Markdown-special_chars_in_code_block** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490903308603-b6028edc0b42-34da97390f31-linux/Markdown-special_chars_in_code_block-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490903308603-b6028edc0b42-34da97390f31-linux/Markdown-special_chars_in_code_block-34da97390f31.png)  
   Markdown-special_chars_in_code_block comparison
 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490903308603-b6028edc0b42-34da97390f31-linux/Markdown-links-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490903308603-b6028edc0b42-34da97390f31-linux/Markdown-links-34da97390f31.png)  
   Markdown-links comparison
 * Changed: **Markdown-Quotes** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490903308603-b6028edc0b42-34da97390f31-linux/Markdown-Quotes-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490903308603-b6028edc0b42-34da97390f31-linux/Markdown-Quotes-34da97390f31.png)  
   Markdown-Quotes comparison
 * Changed: **Markdown-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490903308603-b6028edc0b42-34da97390f31-linux/Markdown-Normal-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490903308603-b6028edc0b42-34da97390f31-linux/Markdown-Normal-34da97390f31.png)  
   Markdown-Normal comparison
 * Changed: **Markdown-Code_block** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490903308603-b6028edc0b42-34da97390f31-linux/Markdown-Code_block-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490903308603-b6028edc0b42-34da97390f31-linux/Markdown-Code_block-34da97390f31.png)  
   Markdown-Code_block comparison
 * Changed: **Markdown-Blank_lines** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490903308603-b6028edc0b42-34da97390f31-linux/Markdown-Blank_lines-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490903308603-b6028edc0b42-34da97390f31-linux/Markdown-Blank_lines-34da97390f31.png)  
   Markdown-Blank_lines comparison
</summary>

File: shared/desktop/webpack.config.production.js [hunk 5]
```diff
@@ -1,9 +1,9 @@
-// @flow
+/* eslint-disable flowtype/require-valid-file-annotation */
```
Comment (chrisnojima) [inline shared/desktop/webpack.config.production.js]: lets not use flow on these files, makes the running of them more of a PITA

File: shared/desktop/webpack.config.production.js [hunk 6]
```diff
@@ -30,6 +30,17 @@ config.plugins.push(
 )
 
 if (!SKIP_OPTIMIZE) {
+  const babelLoader = config.module.loaders.find(l => l.loader === 'babel')
+  const envPreset = babelLoader.query.presets.find(p => p[0] === 'env')[1]
+
+  // Need regenerator
+  babelLoader.query.plugins.push('babel-plugin-transform-runtime')
```
Comment (chrisnojima) [inline shared/desktop/webpack.config.production.js]: Sadly we need to put back all the transpilation so we can minimize

**Comment by chrisnojima** [comment]
- [ ] Actually do a test bot build to sanity check this all

**Comment by chrisnojima** [comment]
@keybase/react-hackers

**Comment by keybase-ci-visdiff** [comment]
The commits b6028edc0b42...6f06faa34b2f introduce visual changes on linux. <details><summary>:mag_right: 6 changed</summary>

 * Changed: **Markdown-special_chars_in_code_block** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490905003499-b6028edc0b42-6f06faa34b2f-linux/Markdown-special_chars_in_code_block-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490905003499-b6028edc0b42-6f06faa34b2f-linux/Markdown-special_chars_in_code_block-6f06faa34b2f.png)  
   Markdown-special_chars_in_code_block comparison
 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490905003499-b6028edc0b42-6f06faa34b2f-linux/Markdown-links-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490905003499-b6028edc0b42-6f06faa34b2f-linux/Markdown-links-6f06faa34b2f.png)  
   Markdown-links comparison
 * Changed: **Markdown-Quotes** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490905003499-b6028edc0b42-6f06faa34b2f-linux/Markdown-Quotes-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490905003499-b6028edc0b42-6f06faa34b2f-linux/Markdown-Quotes-6f06faa34b2f.png)  
   Markdown-Quotes comparison
 * Changed: **Markdown-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490905003499-b6028edc0b42-6f06faa34b2f-linux/Markdown-Normal-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490905003499-b6028edc0b42-6f06faa34b2f-linux/Markdown-Normal-6f06faa34b2f.png)  
   Markdown-Normal comparison
 * Changed: **Markdown-Code_block** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490905003499-b6028edc0b42-6f06faa34b2f-linux/Markdown-Code_block-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490905003499-b6028edc0b42-6f06faa34b2f-linux/Markdown-Code_block-6f06faa34b2f.png)  
   Markdown-Code_block comparison
 * Changed: **Markdown-Blank_lines** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490905003499-b6028edc0b42-6f06faa34b2f-linux/Markdown-Blank_lines-b6028edc0b42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1490905003499-b6028edc0b42-6f06faa34b2f-linux/Markdown-Blank_lines-6f06faa34b2f.png)  
   Markdown-Blank_lines comparison
</summary>

**Comment by chromakode** [comment]
@chrisnojima The visdiff appears to be related to whitespace handling in ```` ```` strings. All of the affected views depict markdown rendered based on fenced strings containing whitspace. Was there a change in the behavior here?

**Comment by chrisnojima** [comment]
don't think so. it looks 100% the same when i run it locally

**Comment by chrisnojima** [comment]
by the same i mean 'correct'

**Comment by chromakode** [comment]
Odd. Perhaps it's something different about the visdiff environment -- node version, or webpack bundle config, etc.

**Pull request merged**